#! /bin/sh /usr/share/dpatch/dpatch-run
## fix-open-or-die.dpatch by Diego Iastrubni <diego.iastrubni@xorcom.com>
##
## DP: Fix some perl syntax issues.

@DPATCH@
diff -urNad freepbx-2.1.3.dfsg/amp_conf/bin/ampbackup.pl /tmp/dpep.hkL9wn/freepbx-2.1.3.dfsg/amp_conf/bin/ampbackup.pl
--- freepbx-2.1.3.dfsg/amp_conf/bin/ampbackup.pl	2005-06-10 21:33:43.000000000 +0300
+++ /tmp/dpep.hkL9wn/freepbx-2.1.3.dfsg/amp_conf/bin/ampbackup.pl	2006-11-26 12:03:32.000000000 +0200
@@ -52,7 +52,7 @@
 # the name of the database our tables are kept
 $database = "asterisk";
 
-open(FILE, "/etc/amportal.conf") || die "Failed to open amportal.conf\n";
+open(FILE, "/etc/amportal.conf") || die "Failed to open amportal.conf (!$)\n";
 while (<FILE>) {
     chomp;                  # no newline
     s/#.*//;                # no comments
diff -urNad freepbx-2.1.3.dfsg/amp_conf/bin/fax-process.pl /tmp/dpep.hkL9wn/freepbx-2.1.3.dfsg/amp_conf/bin/fax-process.pl
--- freepbx-2.1.3.dfsg/amp_conf/bin/fax-process.pl	2006-06-09 01:23:09.000000000 +0300
+++ /tmp/dpep.hkL9wn/freepbx-2.1.3.dfsg/amp_conf/bin/fax-process.pl	2006-11-26 12:03:57.000000000 +0200
@@ -74,7 +74,7 @@
 # Lets make sure that we know about a file...
 die $usage unless $file;
 # and that the file exists...
-open FILE, $file or die "Error opening $file: $!"; 
+open (FILE, $file) or die "Error opening $file: $!"; 
 # Oh, did we possibly not specify an attachment name?
 $attachment = $file unless ($attachment);
 
diff -urNad freepbx-2.1.3.dfsg/amp_conf/bin/retrieve_backup_cron_from_mysql.pl /tmp/dpep.hkL9wn/freepbx-2.1.3.dfsg/amp_conf/bin/retrieve_backup_cron_from_mysql.pl
--- freepbx-2.1.3.dfsg/amp_conf/bin/retrieve_backup_cron_from_mysql.pl	2006-09-11 05:56:34.000000000 +0300
+++ /tmp/dpep.hkL9wn/freepbx-2.1.3.dfsg/amp_conf/bin/retrieve_backup_cron_from_mysql.pl	2006-11-26 12:01:52.000000000 +0200
@@ -40,7 +40,7 @@
 $hostname = $ampconf->{"AMPDBHOST"};
 
 ################### END OF CONFIGURATION #######################
-open(FILE, "/etc/amportal.conf") || die "Failed to open amportal.conf\n";
+open(FILE, "/etc/amportal.conf") or die "Failed to open amportal.conf (!$)\n";
 while (<FILE>) {
     chomp;                  # no newline
     s/#.*//;                # no comments
@@ -58,7 +58,7 @@
 $password = $User_Preferences{"AMPDBPASS"}; 
 
 
-open EXTEN, ">$Backup_cron" or die "Cannot create\/overwrite cron file: $Backup_cron\n";
+open EXTEN, ">$Backup_cron" or die "Cannot create\/overwrite cron file: $Backup_cron (!$)\n";
 
 $dbh = DBI->connect("dbi:mysql:dbname=$database;host=$hostname", "$username", "$password");
 
diff -urNad freepbx-2.1.3.dfsg/amp_conf/bin/retrieve_iax_conf_from_mysql.pl /tmp/dpep.hkL9wn/freepbx-2.1.3.dfsg/amp_conf/bin/retrieve_iax_conf_from_mysql.pl
--- freepbx-2.1.3.dfsg/amp_conf/bin/retrieve_iax_conf_from_mysql.pl	2006-04-25 23:40:47.000000000 +0300
+++ /tmp/dpep.hkL9wn/freepbx-2.1.3.dfsg/amp_conf/bin/retrieve_iax_conf_from_mysql.pl	2006-11-26 12:04:34.000000000 +0200
@@ -67,7 +67,7 @@
   exit;
 }
 
-open EXTEN, ">$iax_conf" or die "Cannot create/overwrite extensions file: $iax_conf\n";
+open (EXTEN, ">$iax_conf") or die "Cannot create/overwrite extensions file: $iax_conf (!$)\n";
 $additional = "";
 my @resultSet = @{$result};
 if ( $#resultSet > -1 ) {
diff -urNad freepbx-2.1.3.dfsg/amp_conf/bin/retrieve_op_conf_from_mysql.pl /tmp/dpep.hkL9wn/freepbx-2.1.3.dfsg/amp_conf/bin/retrieve_op_conf_from_mysql.pl
--- freepbx-2.1.3.dfsg/amp_conf/bin/retrieve_op_conf_from_mysql.pl	2006-11-26 12:00:17.000000000 +0200
+++ /tmp/dpep.hkL9wn/freepbx-2.1.3.dfsg/amp_conf/bin/retrieve_op_conf_from_mysql.pl	2006-11-26 12:02:21.000000000 +0200
@@ -53,7 +53,7 @@
 
 	$zaplabel="Zap \%c";
 	$lastlabelnum=0;
-	open ZAPATA, "<$conffile" || die "Cannot open config file: $zapataconf\n";
+	open( ZAPATA, "<$conffile" ) or die "Cannot open config file: $zapataconf (!$)\n";
 	while( $line = <ZAPATA> ) {
 		next if $line =~ /^(\s)*$/;
 		chomp($line);
@@ -143,7 +143,7 @@
 	$dbh = DBI->connect("dbi:SQLite2:dbname=$db_file","","");
 }
 
-open EXTEN, ">$op_conf" || die "Cannot create/overwrite config file: $op_conf\n";
+open( EXTEN, ">$op_conf" ) or die "Cannot create/overwrite config file: $op_conf (!$)\n";
 
 #First, populate extensions
 
diff -urNad freepbx-2.1.3.dfsg/amp_conf/bin/retrieve_parse_amportal_conf.pl /tmp/dpep.hkL9wn/freepbx-2.1.3.dfsg/amp_conf/bin/retrieve_parse_amportal_conf.pl
--- freepbx-2.1.3.dfsg/amp_conf/bin/retrieve_parse_amportal_conf.pl	2006-02-17 00:40:17.000000000 +0200
+++ /tmp/dpep.hkL9wn/freepbx-2.1.3.dfsg/amp_conf/bin/retrieve_parse_amportal_conf.pl	2006-11-26 12:00:17.000000000 +0200
@@ -11,7 +11,7 @@
 	my $filename = $_[0];
 	my %ampconf;
 	
-	open(AMPCONF, $filename) || die "Cannot open $filename";
+	open(AMPCONF, $filename) or die "Cannot open $filename (!$)";
 	
 	while (<AMPCONF>)
 	{
diff -urNad freepbx-2.1.3.dfsg/amp_conf/bin/retrieve_queues_conf_from_mysql.pl /tmp/dpep.hkL9wn/freepbx-2.1.3.dfsg/amp_conf/bin/retrieve_queues_conf_from_mysql.pl
--- freepbx-2.1.3.dfsg/amp_conf/bin/retrieve_queues_conf_from_mysql.pl	2006-03-06 17:07:41.000000000 +0200
+++ /tmp/dpep.hkL9wn/freepbx-2.1.3.dfsg/amp_conf/bin/retrieve_queues_conf_from_mysql.pl	2006-11-26 12:02:43.000000000 +0200
@@ -62,7 +62,7 @@
   exit;
 }
 
-open EXTEN, ">$queues_conf" or die "Cannot create/overwrite extensions file: $queues_conf\n";
+open EXTEN, ">$queues_conf" or die "Cannot create/overwrite extensions file: $queues_conf (!$)\n";
 $additional = "";
 my @resultSet = @{$result};
 if ( $#resultSet > -1 ) {
diff -urNad freepbx-2.1.3.dfsg/amp_conf/bin/retrieve_sip_conf_from_mysql.pl /tmp/dpep.hkL9wn/freepbx-2.1.3.dfsg/amp_conf/bin/retrieve_sip_conf_from_mysql.pl
--- freepbx-2.1.3.dfsg/amp_conf/bin/retrieve_sip_conf_from_mysql.pl	2006-04-25 23:40:47.000000000 +0300
+++ /tmp/dpep.hkL9wn/freepbx-2.1.3.dfsg/amp_conf/bin/retrieve_sip_conf_from_mysql.pl	2006-11-26 12:02:54.000000000 +0200
@@ -66,7 +66,7 @@
 	exit;
 }
 
-open EXTEN, ">$sip_conf" or die "Cannot create/overwrite extensions file: $sip_conf\n";
+open EXTEN, ">$sip_conf" or die "Cannot create/overwrite extensions file: $sip_conf (!$)\n";
 $additional = "";
 my @resultSet = @{$result};
 if ( $#resultSet > -1 ) {
diff -urNad freepbx-2.1.3.dfsg/amp_conf/bin/retrieve_zap_conf_from_mysql.pl /tmp/dpep.hkL9wn/freepbx-2.1.3.dfsg/amp_conf/bin/retrieve_zap_conf_from_mysql.pl
--- freepbx-2.1.3.dfsg/amp_conf/bin/retrieve_zap_conf_from_mysql.pl	2006-04-25 23:40:47.000000000 +0300
+++ /tmp/dpep.hkL9wn/freepbx-2.1.3.dfsg/amp_conf/bin/retrieve_zap_conf_from_mysql.pl	2006-11-26 12:03:13.000000000 +0200
@@ -66,7 +66,7 @@
 	exit;
 }
 
-open EXTEN, ">$zap_conf" or die "Cannot create/overwrite extensions file: $zap_conf\n";
+open EXTEN, ">$zap_conf" or die "Cannot create/overwrite extensions file: $zap_conf ($!)\n";
 $additional = "";
 my @resultSet = @{$result};
 if ( $#resultSet > -1 ) {
